android/src/main/java/com/pspdfkit/flutter/pspdfkit/ConfigurationAdapter.java

```diff
@@ -0,0 +1,370 @@
+        if (enableAnnotationEditing) {
+            configuration.enableAnnotationEditing();
+        } else {
+            configuration.disableAnnotationEditing();
+        }
+    }
+
+    private void configureShowShareAction(boolean showShareAction) {
+        if (showShareAction) {
+            configuration.setEnabledShareFeatures(ShareFeatures.all());
+        } else {
+            configuration.setEnabledShareFeatures(ShareFeatures.none());
+        }
+    }
+
+    private void configureShowPrintAction(boolean showPrintAction) {
+        if (showPrintAction) {
+            configuration.enablePrinting();
+        } else {
+            configuration.disablePrinting();
+        }
+    }
+
+    private void configureEnableTextSelection(boolean enableTextSelection) {
+        configuration.textSelectionEnabled(enableTextSelection);
+    }
+
+    private void configureDocumentInfoView(boolean enableDocumentInfoView) {
+        if (enableDocumentInfoView) {
+            configuration.enableDocumentInfoView();
+        } else {
+            configuration.disableDocumentInfoView();
+        }
+    }
+
+    private void configureThemeMode(String themeMode) {
+        ThemeMode result = ThemeMode.DEFAULT;
+        if (themeMode.equals(APPEARANCE_MODE_DEFAULT)) {
+            result = ThemeMode.DEFAULT;
+        } else if (themeMode.equals(APPEARANCE_MODE_NIGHT)) {
+            result = ThemeMode.NIGHT;
+        }
+        configuration.themeMode(result);
+    }
+
+    private void configureDarkThemeRes(String darkThemeResource, Context context) {
+        @StyleRes int darkThemeId = getStyleResourceId(darkThemeResource, context);
+        if (darkThemeId != 0) {
+            configuration.themeDark(darkThemeId);
+        }
+    }
+
+    private void configureDefaultThemeRes(String defaultThemeResource, Context context) {
+        @StyleRes int defaultThemeId = getStyleResourceId(defaultThemeResource, context);
+        if (defaultThemeId != 0) {
+            configuration.theme(defaultThemeId);
+        }
+    }
+
+    private static int getStyleResourceId(String styleName, Context context) {
+        int resourceId = context.getResources().getIdentifier(styleName, "style", context.getPackageName());
+        if (resourceId == 0) {
+            Log.e(LOG_TAG, String.format("Style resource not found for %s", styleName));
+        }
+        return resourceId;
+    }
+
+    private <T> boolean containsKeyOfType(HashMap<String, Object> configurationMap, String key, Class<T> clazz) {
+        if (configurationMap.get(key) != null) {
+            checkCast(configurationMap.get(key), clazz, key);
+            return true;
+        }
+        return false;
+    }
+
+    private static <T> void checkCast(Object object, Class<T> clazz, String key) {
+        if (!clazz.isInstance(object)) {
+            throw new ClassCastException(String.format("Value for the key %s must be of type %s.", key, clazz.getName()));
```

Comment:
This will show the Java name of the type in the exception. Wouldn't it make sense to instead use the Dart type that is expected to be used?

Reply:
Yes, good idea!